src/taxonomy/taxonomy-card/TaxonomyCardMenu.jsx

```diff
@@ -29,16 +29,28 @@ const TaxonomyCardMenu = ({
         alt={intl.formatMessage(messages.taxonomyMenuAlt, { name })}
         id={`taxonomy-card-menu-button-${id}`}
         data-testid={`taxonomy-card-menu-button-${id}`}
+        disabled={disabled}
       />
       <Dropdown.Menu data-testid={`taxonomy-card-menu-${id}`}>
         {/* Add more menu items here */}
-        <Dropdown.Item
-          className="taxonomy-menu-item"
-          data-testid={`taxonomy-card-menu-export-${id}`}
-          onClick={(e) => onClickItem(e, 'export')}
-        >
-          {intl.formatMessage(messages.taxonomyCardExportMenu)}
-        </Dropdown.Item>
+        { menuItems.includes('export') && (
+          <Dropdown.Item
+            className="taxonomy-menu-item"
+            data-testid={`taxonomy-card-menu-export-${id}`}
+            onClick={(e) => onClickItem(e, 'export')}
+          >
+            {intl.formatMessage(messages.taxonomyCardExportMenu)}
+          </Dropdown.Item>
+        )}
+        { menuItems.includes('delete') && (
+          <Dropdown.Item
+            className="taxonomy-menu-item"
+            data-testid={`taxonomy-card-menu-delete-${id}`}
+            onClick={(e) => onClickItem(e, 'delete')}
+          >
+            {intl.formatMessage(messages.taxonomyCardDeleteMenu)}
+          </Dropdown.Item>
+        )}
```

Comment:
Instead of this why not just use `menuIems.map`?
```
menuItems.map(item => (
<Dropdown.Item
  className="taxonomy-menu-item"
onClick = {onClickItem(item)}
>
{intl.formatMessage(menuMessages[item])}
</Dropdown.Item>
)
```
onClickItem can be changed to:

```javascript
const onClickItem = (menuName) => (e) =>{
    e.preventDefault();
    onClickMenuItem(menuName);
  };
```

Reply:
Updated here: https://github.com/openedx/frontend-app-course-authoring/pull/684/commits/a4fc2ac7d4c9f14aac2df119f0ca5e1b06a3a4e0